sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/TestClassShouldHaveTestMethod.cs

```diff
@@ -49,14 +49,10 @@ public sealed class TestClassShouldHaveTestMethod : SonarDiagnosticAnalyzer
         };
 
         private static readonly ISet<KnownType> HandledTestMethodAttributes = new HashSet<KnownType>
-        {
-            KnownType.Microsoft_VisualStudio_TestTools_UnitTesting_TestMethodAttribute,
-            KnownType.Microsoft_VisualStudio_TestTools_UnitTesting_DataTestMethodAttribute,
-            KnownType.NUnit_Framework_TestAttribute,
-            KnownType.NUnit_Framework_TestCaseAttribute,
-            KnownType.NUnit_Framework_TestCaseSourceAttribute,
-            KnownType.NUnit_Framework_TheoryAttribute
-        };
+        (
+            UnitTestHelper.KnownTestMethodAttributes_MSTest
+            .Concat(UnitTestHelper.KnownTestMethodAttributes_NUnit)
+        );

```

Comment:
Can't use _UnitTestHelper.IsTestMethod_ here since rule doesn't apply to _xUnit_.
I considered adding separate helper methods for the various test frameworks, but it didn't seem worth it. Instead, I've exposed the commonly-used groups of attributes in the _UnitTestHelper_ so they only need to be updated in one place.

Reply:
The rule does not apply to XUnit because it does not have equivalent to the [TestClass] attribute and all XUnit test classes already have tests by definition. Imo it does not hurt if we use the general IsTestMethod even though a few of the checked attributes will be redundant...

Reply:
Ok. I'll move the "IsTestClass" code into _UnitTestHelper_ too.